trieste/acquisition/rule.py

```diff
@@ -1013,6 +1013,11 @@ def __init__(
         self.region_index = region_index
         self.input_active_dims = input_active_dims
 
+    @property
+    def needs_init(self) -> bool:
+        """Return `True` if the search space needs to be initialized, and `False` otherwise."""
```

Comment:
Maybe expand here a bit?

```suggestion
        """Return `True` if the search space needs to be re-initialized with the latest datasets, and `False` if it can be just updated."""
```